apps/aecore/src/aec_chain_state.erl

```diff
@@ -426,18 +432,19 @@ internal_insert(Node, Block, Origin) ->
             %% Only add the block if we can do the whole
             %% transitive operation (i.e., calculate all the state
             %% trees, and update the pointers)
-            Fun = fun() -> internal_insert_transaction(Node, Block, Origin)
+            Fun = fun() ->
+                          internal_insert_transaction(Node, Block, Origin)
                   end,
             try
                 aec_db:ensure_transaction(Fun)
             catch
                 exit:{aborted, {throw, ?internal_error(What)}} ->
-                    internal_error(What)
+                    {error, What}
```

Comment:
that's a big change. internal_insert is not exported - OK.

Reply:
Yes. The previous setup would double throw the error. Since only this tagged error was caught anyway, we can get rid of the second throw entirely.